lib/kubernetes-deploy/runner.rb

```diff
@@ -106,6 +108,13 @@ def run
       phase_heading("Checking initial resource statuses")
       resources.each(&:sync)
 
+      ejson = EjsonSecretProvisioner.new(namespace: @namespace, context: @context, template_dir: @template_dir)
+      if ejson.secrets_requested?
```

Comment:
I did it this way to be able to avoid printing the heading when there's nothing to do, i.e. hide the feature from the average deploy user. 

> Also what if someone just deletes the file, then we never prune the secrets?

Good point. `secrets_requested?` would need to be `secret_changes_requested?` and look at live annotated secrets as well. I should add a test for this regardless of the heading issue.